foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/utils/RSAUtils.java

```diff
@@ -0,0 +1,63 @@
+package io.servicecomb.foundation.common.utils;
+
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+public class RSAUtils {
+
+  private final static String RSA_ALG = "RSA";
+
+  private final static String SIGN_ALG = "SHA256withRSA";
+  
+  private final static int KEY_SIZE = 2048;
+
+  private static Base64.Encoder encoder = Base64.getEncoder();
+
+  private static Base64.Decoder decoder = Base64.getDecoder();
+
+  public static RSAKeyPairEntry getRSAKeyPair() {
+    try {
+      KeyPairGenerator keyGenerator = KeyPairGenerator.getInstance(RSA_ALG);
+      keyGenerator.initialize(KEY_SIZE, new SecureRandom());
+      KeyPair keyPair = keyGenerator.generateKeyPair();
+      PublicKey pubKey = keyPair.getPublic();
+      PrivateKey privKey = keyPair.getPrivate();
+      return new RSAKeyPairEntry(privKey, pubKey, encoder.encodeToString(pubKey.getEncoded()));
+    } catch (NoSuchAlgorithmException e) {
+      throw new Error(e);
```

Comment:
Avoid using Error, when you need throw some unexpected exception, using RuntimeException, e.g. IllegalStageException or IllgaleArgumentException

Reply:
It's better to write a log for it.

Reply:
accept